test/js/node/process/process.test.js

```diff
@@ -377,6 +377,30 @@ describe("process.onBeforeExit", () => {
     expect(exitCode).toBe(0);
     expect(stdout.toString().trim()).toBe("beforeExit: 0\nbeforeExit: 1\nexit: 2");
   });
+
+  it("throwing inside preserves exit code", async () => {
```

Comment:
* what if the exit code is non-0 and non-1? does it become 1 or keep the non-1?
* what if the error is thrown in a signal handler?

Reply:
this exits with 4 in both node and bun, i believe this path is only for when the exit is otherwise clean

```js
process.exitCode = 4;
process.on("exit", code => {
  console.log("process", "exit", code, process.exitCode);
  throw new Error("boom");
});
```

if u find a repro where the exit code is clobbered and differs from node im happy to look into it